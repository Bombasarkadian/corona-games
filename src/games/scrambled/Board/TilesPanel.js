import React from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import { Label, Button, Segment, Icon, Sticky } from "semantic-ui-react";

import Tile from "../Tile";

const propTypes = {
  t: PropTypes.func.isRequired,
  stickyRef: PropTypes.object,
  disabled: PropTypes.bool,
  playerTiles: PropTypes.arrayOf(PropTypes.object),
  selectTile: PropTypes.func.isRequired,
  selectedTile: PropTypes.object,
  onReturnTiles: PropTypes.func.isRequired,
  onSwapTiles: PropTypes.func.isRequired,
  onSkipTurn: PropTypes.func.isRequired,
  onPlayTiles: PropTypes.func.isRequired,
  canReturnTiles: PropTypes.bool,
  canPlayTiles: PropTypes.bool,
};

const TilesPanel = ({
  t,
  stickyRef,
  disabled,
  playerTiles,
  selectTile,
  selectedTile,
  onReturnTiles,
  onSwapTiles,
  onSkipTurn,
  onPlayTiles,
  canReturnTiles,
  canPlayTiles,
}) => (
  <Sticky context={stickyRef}>
    <Segment disabled={disabled}>
      <Label as="span" color="green" ribbon>
        <Icon name="star" />
        {t("panel.tiles")}
      </Label>
      <div style={{ display: "flex", paddingTop: 10 }}>
        {playerTiles &&
          playerTiles.map((tile) => (
            <Tile
              separate
              key={tile.id}
              onClick={() => selectTile(tile)}
              raised={selectedTile === tile}
              disabled={tile.x !== undefined && tile.y !== undefined}
              {...tile}
            />
          ))}
        <Button color="red" onClick={onReturnTiles} disabled={disabled || !canReturnTiles}>
          <Icon name="undo" />
          {t("panel.actions.return")}
        </Button>
        <Button color="orange" onClick={onSwapTiles} disabled={disabled}>
          <Icon name="refresh" />
          {t("panel.actions.swap")}
        </Button>
        <Button color="yellow" onClick={onSkipTurn} disabled={disabled}>
          <Icon name="forward" />
          {t("panel.actions.skip")}
        </Button>
        <Button color="green" onClick={onPlayTiles} disabled={disabled || !canPlayTiles}>
          <Icon name="check" />
          {t("panel.actions.play")}
        </Button>
      </div>
    </Segment>
  </Sticky>
);

TilesPanel.propTypes = propTypes;
export default withTranslation("scrambled")(TilesPanel);
