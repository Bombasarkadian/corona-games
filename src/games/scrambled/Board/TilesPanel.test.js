import React from "react";
import renderer from "react-test-renderer";
import TilesPanel from "./TilesPanel";

describe("scrambled TilesPanel component", () => {
  it("renders correctly", () => {
    expect.hasAssertions();
    const component = renderer.create(
      <TilesPanel
        t={jest.fn((x) => x)}
        playerTiles={[{ letter: "A", points: 1, id: 1 }]}
        selectTile={jest.fn()}
        onReturnTiles={jest.fn()}
        onSwapTiles={jest.fn()}
        onSkipTurn={jest.fn()}
        onPlayTiles={jest.fn()}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
