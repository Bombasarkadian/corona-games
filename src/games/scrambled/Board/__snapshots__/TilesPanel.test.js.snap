// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrambled TilesPanel component renders correctly 1`] = `
<div
  className=""
>
  <div
    style={Object {}}
  />
  <div
    className="ui sticky"
  >
    <div
      className="ui segment"
    >
      <span
        className="ui green ribbon label"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="star icon"
          onClick={[Function]}
        />
        panel.tiles
      </span>
      <div
        style={
          Object {
            "display": "flex",
            "paddingTop": 10,
          }
        }
      >
        <div
          className="tile tile_separate tile_clickable"
          onClick={[Function]}
        >
          <div
            className="letter"
          >
            A
          </div>
          <div
            className="points"
          >
            1
          </div>
        </div>
        <button
          className="ui red disabled button"
          disabled={true}
          onClick={[Function]}
          tabIndex={-1}
        >
          <i
            aria-hidden="true"
            className="undo icon"
            onClick={[Function]}
          />
          panel.actions.return
        </button>
        <button
          className="ui orange button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="refresh icon"
            onClick={[Function]}
          />
          panel.actions.swap
        </button>
        <button
          className="ui yellow button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="forward icon"
            onClick={[Function]}
          />
          panel.actions.skip
        </button>
        <button
          className="ui green disabled button"
          disabled={true}
          onClick={[Function]}
          tabIndex={-1}
        >
          <i
            aria-hidden="true"
            className="check icon"
            onClick={[Function]}
          />
          panel.actions.play
        </button>
      </div>
    </div>
  </div>
</div>
`;
